src/dataflow/DataflowInferenceTreeAnnotator.java

```diff
@@ -105,13 +106,39 @@ public Void visitMethodInvocation(MethodInvocationTree methodInvocationTree,
                     .put(tm.toString(), tm);
             AnnotationMirror anno = DataflowUtils.genereateDataflowAnnoFromByteCode(atm,
                     this.realTypeFactory.getProcessingEnv());
+
+            if (atm instanceof AnnotatedArrayType) {
```

Comment:
You shouldn't use instanceof to separate AnnotatedTypeMirror. Use `.getKind()` and compare to the TypeKind.